lib/p5.play.js

```diff
@@ -1546,14 +1546,14 @@ function Sprite(pInst, _x, _y, _w, _h) {
     get: function() {
       if (this._internalWidth === undefined) {
         return 100;
-      } else if (this.animation) {
+      } else if (this.animation && pInst._fixedSpriteAnimationFrameSizes) {
```

Comment:
Interesting - in these cases, is `_fixedSpriteAnimationFrameSizes` essentially a proxy for "Use the Code.org logic?"  If so, it might be okay to go ahead and assume this is true in our fork.

You've added good `setAnimation` tests, but I don't see this new property in any of the test code (and no existing tests failing because of it) which suggests you've found a spot missing test coverage.  If we're going to introduce the property I'd like us to add some tests for its effect... which might be an even better reason to just assume `true` in our fork!

Reply:
The interesting thing is that there are many test cases that break when this mode is enabled (e.g. https://github.com/code-dot-org/p5.play/blob/master/test/unit/axis-aligned-bounding-box.js#L82 ). I debated this for a while. In the end, I decided to preserve both modes because I can see validity to either mode. I think I'd rather just add new test cases to verify our mode behaves as we expect. @islemaste, do you have any recollection on why we changed this behavior? I assume it was a simplification related to sprite sheets and how we wanted the piskel editor to work?

Reply:
That's correct.  As written, a sprite sheet instance in p5.play can have different dimensions for every frame (although the actual support for this is poor). In practice this is rarely a good idea and makes certain calculations - especially some of the additional squash/stretch behavior we've added - terribly difficult to get right for all cases.  In addition, Piskel assumes all frames in an animation will be the same size, so it made sense to go ahead and introduce this constraint in the engine.

Keeping both modes and adding corresponding tests is fine by me, but within our application I think this mode will always be on.

Reply:
👍